python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -462,6 +462,11 @@ def write(self, value: Any, timeout: Optional[float] = None) -> None:
         # -1 means no timeout (block indefinitely)
         timeout_ms = int(timeout * 1000) if timeout is not None else -1
 
+        from ray.experimental.channel.gpu_communicator import GPUFuture
+
+        if isinstance(value, GPUFuture):
+            value = value.wait()
```

Comment:
Ideally we should remove all cases of `if isinstance(value, GPUFuture)` and instance standardize all interfaces to take a generic `Future` object.

Reply:
Done.